components/browser/state/src/test/java/mozilla/components/browser/state/action/WebExtensionActionTest.kt

```diff
@@ -249,10 +251,25 @@ class WebExtensionActionTest {
         store.dispatch(WebExtensionAction.InstallWebExtensionAction(extension)).joinBlocking()
 
         assertEquals(extension, store.state.extensions[extension.id])
-        assertNull(store.state.extensions[extension.id]?.browserActionPopupSession)
+        assertNull(store.state.extensions[extension.id]?.popupSessionId)
 
         store.dispatch(WebExtensionAction.UpdatePopupSessionAction(extension.id, "popupId")).joinBlocking()
-        assertEquals("popupId", store.state.extensions[extension.id]?.browserActionPopupSession)
+        assertEquals("popupId", store.state.extensions[extension.id]?.popupSessionId)
+    }
+
+    @Test
+    fun `UpdateActionPopupSession - Adds popup engine session to the web extension state`() {
+        val store = BrowserStore()
+
+        val extension = WebExtensionState("id", "url")
+        store.dispatch(WebExtensionAction.InstallWebExtensionAction(extension)).joinBlocking()
+
+        assertEquals(extension, store.state.extensions[extension.id])
+        assertNull(store.state.extensions[extension.id]?.popupSessionId)
```

Comment:
If you want a separate test for the two states (we could also combine the tests), then this should be:
` assertNull(store.state.extensions[extension.id]?.popupSession)`